dali/python/nvidia/dali/data_node.py

```diff
@@ -259,6 +260,32 @@ def process_index(idx, dim):
         else:
             return nvidia.dali.fn.expand_dims(sliced, axes=new_axes, new_axis_names=new_axis_names)
 
+    def shape(self, *, dtype=None, device="cpu"):
+        """Returns the run-time shapes of this DataNode as a new DataNode
+
+        Parameters
+        ----------
+        arg_dtype : DALIDataType, optional
+            If specified, the shape will be converted to this data type; defaults to INT64.
+        device : str, optional
+            The device ("cpu" or "gpu") where the result is returned; defaults to CPU.
+        """
+        from . import fn
+
+        if device == "cpu":
+            self._check_gpu2cpu()
+        return fn.shapes(self, dtype=dtype, device=device)
+
+    def _check_gpu2cpu(self):
```

Comment:
Can you add a docstring here where this should be used? And maybe a warning why we don't use current_pipeline.